server/jetstream_cluster_2_test.go

```diff
@@ -6361,24 +6361,28 @@ func TestJetStreamClusterEncryptedDoubleSnapshotBug(t *testing.T) {
 	require_NoError(t, err)
 }
 
-func TestJetStreamClusterRePublishUpdateNotSupported(t *testing.T) {
+func TestJetStreamClusterRePublishUpdateSupported(t *testing.T) {
 	test := func(t *testing.T, s *Server, stream string, replicas int) {
 		nc := natsConnect(t, s.ClientURL())
 		defer nc.Close()
 
 		cfg := &StreamConfig{
-			Name:     stream,
+			Name:     stream + "_1",
 			Storage:  MemoryStorage,
 			Replicas: replicas,
+			Subjects: []string{"foo.>"},
 		}
 		addStream(t, nc, cfg)
 
-		cfg.RePublish = &RePublish{
-			Source:      ">",
-			Destination: "bar.>",
+		cfg2 := &StreamConfig{
+			Name:     stream + "_2",
+			Storage:  MemoryStorage,
+			Replicas: replicas,
+			Subjects: []string{"bar.>"},
 		}
-		// We expect update to fail, do it manually:
-		expectFailUpdate := func() {
+		addStream(t, nc, cfg2)
```

Comment:
Can use js.AddStream(*cfg) now, this test seems a bit older and was using old style.